indico/modules/events/abstracts/models/abstracts.py

```diff
@@ -82,54 +169,136 @@ def __table_args__(cls):
+    submitted_for_tracks = db.relationship(
+        'Track',
+        secondary='event_abstracts.submitted_for_tracks',
+        collection_class=set,
+        backref=db.backref(
+            'abstracts_submitted',
             lazy=True
         )
     )
+    proposed_for_tracks = db.relationship(
+        'Track',
+        secondary='event_abstracts.proposed_for_tracks',
+        collection_class=set,
+        backref=db.backref(
+            'abstracts_proposed'
+        )
+    )
     type = db.relationship(
         'ContributionType',
         lazy=True,
         foreign_keys=[type_id],
         backref=db.backref(
-            'abstracts',
+            'proposed_abstracts',
             lazy=True
         )
     )
-    accepted_type = db.relationship(
+    #: User who judged the abstract
+    judge = db.relationship(
+        'User',
+        lazy=True,
+        foreign_keys=judge_id,
+        backref=db.backref(
+            'judged_abstracts',
+            cascade='all, delete-orphan',
+            lazy='dynamic'
+        )
+    )
+    final_type = db.relationship(
         'ContributionType',
         lazy=True,
-        foreign_keys=[accepted_type_id],
+        foreign_keys=[final_type_id],
+        backref=db.backref(
+            'completed_abstracts',
+            lazy=True
+        )
+    )
+    final_track = db.relationship(
+        'Track',
+        lazy=True,
+        foreign_keys=[final_track_id],
+        backref=db.backref(
+            'abstracts_accepted',
+            lazy=True
+        )
+    )
+    merged_into = db.relationship(
+        'Abstract',
+        lazy=True,
+        remote_side=id,
+        foreign_keys=merged_into_id,
+        backref=db.backref(
+            'merged_abstracts',
+            lazy=True
+        )
+    )
+    duplicate_of = db.relationship(
+        'Abstract',
+        lazy=True,
+        remote_side=id,
+        foreign_keys=duplicate_of_id,
+        backref=db.backref(
+            'dupe_abstracts',
+            lazy=True
+        )
+    )
+    #: Data stored in abstract/contribution fields
+    field_values = db.relationship(
+        'AbstractFieldValue',
+        lazy=True,
+        cascade='all, delete-orphan',
         backref=db.backref(
-            'accepted_as_abstracts',
+            'abstract',
             lazy=True
         )
     )
+
     # relationship backrefs:
     # - contribution (Contribution.abstract)
-    # - judgments (Judgment.abstract)
+    # - dupe_abstracts (Abstract.dupe_of)
+    # - email_logs (AbstractEmailLog.abstract)
+    # - merged_abstracts (Abstract.merged_into)
+    # - reviews (AbstractReview.abstract)
+
+    @property
+    def public_state(self):
+        if self.state != AbstractState.submitted:
+            return self.state
+        elif self.reviews:
+            return AbstractPublicState.under_review
+        else:
+            return AbstractPublicState.awaiting
+
+    @property
+    def reviewing_state(self):
+        if not self.reviews:
```

Comment:
That's not true. Somebody might have proposed another track, duplicate or merge.